<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Data Preprocessing Module</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_main">
    <item>
     <widget class="QLabel" name="label_title">
      <property name="font">
       <font>
        <pointsize>24</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Data Preprocessing Module</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_raw_data">
      <item>
       <widget class="QLabel" name="label_raw_data">
        <property name="text">
         <string>Raw Data Folder:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_raw_data"/>
      </item>
      <item>
       <widget class="QPushButton" name="button_browse_raw_data">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_label_files">
      <item>
       <widget class="QLabel" name="label_label_files">
        <property name="text">
         <string>Label Files Folder:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_label_files"/>
      </item>
      <item>
       <widget class="QPushButton" name="button_browse_label_files">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="preprocessingGroupBox">
      <property name="title">
       <string>Preprocessing Module</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_preprocessing">
       <item>
        <widget class="QCheckBox" name="downsampleCheckBox">
         <property name="text">
          <string>Downsample Point Cloud</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_downsample">
         <item>
          <widget class="QLabel" name="downsampleLabel">
           <property name="text">
            <string>Every K Points:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="downsampleSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="value">
            <number>8</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="removeOutlierCheckBox">
         <property name="text">
          <string>Remove Statistical Outlier</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_removeOutlier">
         <item>
          <widget class="QLabel" name="nbNeighborsLabel">
           <property name="text">
            <string>Neighbors:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="nbNeighborsSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="stdRatioLabel">
           <property name="text">
            <string>Std Ratio:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="stdRatioDoubleSpinBox">
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="roiCroppingCheckBox">
         <property name="text">
          <string>ROI Cropping</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_roiCropping">
         <item>
          <widget class="QLabel" name="roiCroppingLabel">
           <property name="text">
            <string>X Range:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="roiCroppingSpinBox">
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="value">
            <double>5.130000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_controls">
      <item>
       <widget class="QPushButton" name="button_start">
        <property name="text">
         <string>Start Preprocessing</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_reset">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_finish">
        <property name="text">
         <string>Finish</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTextEdit" name="textEdit_log">
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="acceptRichText">
       <bool>false</bool>
      </property>
      <property name="placeholderText">
       <string>Logs will be displayed here...</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
